### PR TITLE
JS3 week1 homework

### PR DESCRIPTION
**Pull request by shpomp**

### CODE REVIEW

File: javascript/javascript3/week1/giphy/giphy.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    }
+    else {
+        fetchURL = buildURL(searchKeyword);
+        triggerSearchButton.innerText = "search again!";
+    }
+    fetchGifFunction(fetchURL);
+  };
+
+  function buildURL(keyword){
+     const URL = `https://api.giphy.com/v1/gifs/search?api_key=z7JV6Nh4HDwegGaxm3Zn65dthkpODSvy&q=${keyword}&limit=50&offset=0&rating=g&lang=en`;
```
Comment (valmdz) [inline javascript/javascript3/week1/giphy/giphy.js:58]: The URL has the limit option query as the case of the keyword, might be good to use it and in that way control the amount of gifs shown as it is currently fetching 50 results every time the function is called ;)

File: javascript/javascript3/week1/giphy/giphy.html [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <link rel="stylesheet" href="giphy.css" />
+    <title>Document</title>
+  </head>
+  <body>
+    <div id="search-div">
+      <h1>find a gif!</h1>
+      <input
```
Comment (valmdz) [inline javascript/javascript3/week1/giphy/giphy.html:13]: It is good to have the `input `fields and the `button `inside a form and then update the event target in line 62 in giphy.js to `submit `

File: javascript/javascript3/week1/giphy/giphy.js (hunk 1 above)
Comment (valmdz) [reply javascript/javascript3/week1/giphy/giphy.js:58]: `BuildURL `is called before it's definition, might be good to instantiate it at the beginning of the code

File: javascript/javascript3/week1/giphy/giphy.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+// https://api.giphy.com/v1/gifs/search?api_key=z7JV6Nh4HDwegGaxm3Zn65dthkpODSvy&q=tree&limit=3&offset=0&rating=g&lang=en
+
+// q=tree is the searchword
+// limit=3 is the amount of gifs
+
+const triggerSearchButton = document.getElementById("trigger-search-button");
+const searchResultsDiv = document.getElementById("search-results-div");
+const message = document.createElement("h2");
+let fetchURL = "";
```
Comment (valmdz) [inline javascript/javascript3/week1/giphy/giphy.js:11]: will `fetchURL `always be the empty string?

**Review by valmdz — COMMENTED** [review]
It seems good overall, Karo :)
a linter might be good to remove extra lines ;)

File: javascript/javascript3/week1/giphy/giphy.js (hunk 1 above)
Comment (shpomp) [reply javascript/javascript3/week1/giphy/giphy.js:58]: I was using the limit option in URL initially,  but got bored of seeing the same gifs every time, and got an idea to try to implement some randomness, that is why it is like that :)

File: javascript/javascript3/week1/giphy/giphy.js (hunk 3 above)
Comment (shpomp) [reply javascript/javascript3/week1/giphy/giphy.js:11]: I might be having leftover effects from Java. Obviously, this initialization is not needed at all. 🤦🏽

File: javascript/javascript3/week1/giphy/giphy.html (hunk 2 above)
Comment (shpomp) [reply javascript/javascript3/week1/giphy/giphy.html:13]: adjusting just that breaks the search, I will leave it for now due to time constraints but will keep it in mind for the next time!

**Pull request merged**